### DISCUSSION TITLE
How can add a new chain to the WallectconnectV2-web3?

### DISCUSSION THREAD

**Discussion by DonJ92** [WalletConnect/walletconnect-monorepo#1850] [Developer Support]
I'm testing WallectconnectV2 by demo([react-dapp-v2-with-web3js](https://github.com/WalletConnect/web-examples/tree/main/dapps/react-dapp-v2-with-web3js))
And I want to add a new custom chain based ethereum.
So I add chain info to below files.
https://github.com/WalletConnect/web-examples/blob/main/dapps/react-dapp-v2-with-web3js/src/chains/eip155.ts
https://github.com/WalletConnect/web-examples/blob/main/dapps/react-dapp-v2-with-web3js/src/constants/default.ts

But this example is using ChainData from [caip-api ](https://github.com/pedrouid/caip-api) and I have to add chain info to caip-api.
https://github.com/WalletConnect/web-examples/blob/e4fdfc63377d53354d0ceefe14faa2d3efb8440b/dapps/react-dapp-v2-with-web3js/src/components/Blockchain.tsx#L3
https://github.com/WalletConnect/web-examples/blob/e4fdfc63377d53354d0ceefe14faa2d3efb8440b/dapps/react-dapp-v2-with-web3js/src/contexts/ClientContext.tsx#L12

**Comment by DonJ92** [comment]
@pedrouid @bkrem @ganchoradkov @crypblizz8 
I'm sorry but please check this discussion.

**Reply by aaronmgdr** [reply]
see https://github.com/orgs/WalletConnect/discussions/1769#discussioncomment-4844416
